users/tests.py

```diff
@@ -1,3 +1,117 @@
-# from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.authtoken.models import Token
+from rest_framework.test import APITestCase
 
-# Create your tests here.
+from users.models import User
+
+
+class UserTest(APITestCase):
+    def setUp(self):
+        self.signup_data = {"username": "harrypotter", "password": "hogwarts934"}
+        self.client.post(reverse("signup"), self.signup_data)
+        self.data = {
+            "first_name": "",
+            "last_name": "",
+            "description": None,
+            "username": "harrypotter",
+            "date_of_birth": None,
+            "profile_photo": None,
+            "id": User.objects.get(username="harrypotter").id,
+            "email": "",
+        }
+
+        self.signup_data_another = {
+            "username": "robinsoncrusoe",
+            "password": "daniel1719",
+        }
+        self.client.post(reverse("signup"), self.signup_data_another)
+
+    def get_argument(self, username):
+        return {"pk": User.objects.get(username=username).id}
+
+    def authenticate(self, login_data):
+        response = self.client.post(reverse("token"), login_data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        user = User.objects.get(username=login_data.get("username"))
+        token = Token.objects.get(user=user)
+        self.assertEqual(response.data["token"], token.key)
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + token.key)
+
+
+class CreateUserTest(APITestCase):
+    def setUp(self):
+        self.signup_data = {"username": "harrypotter", "password": "hogwarts934"}
+        self.signup_data_another = {"username": "robinsoncrusoe", "password": "123"}
+
+    def test_can_create_user(self):
+        response = self.client.post(reverse("signup"), self.signup_data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_can_not_create_user(self):
+        response = self.client.post(reverse("signup"), self.signup_data_another)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        self.test_can_create_user()
```

Comment:
Not sure it's a good idea to call one test from another as it may change and broke the current one